gn/BUILD.gn

```diff
@@ -364,16 +409,23 @@ config("warnings") {
 
       "-Wno-deprecated-declarations",
       "-Wno-maybe-uninitialized",
-      "-Wno-extra-semi-stmt",
     ]
     cflags_cc += [
       "-Wnon-virtual-dtor",
       "-Wno-noexcept-type",
+
+      # GCC 8+ bundles a number of fundamentally different warnings under this same flag,
+      # ranging from false positive (only copying a prefix of SkRRect from serialized form)
+      # to possibly useful (memcpy() with non-trivial types).  Annoyingly you can't really
+      # break them up any finer.
+      # TODO(mtklein): suppress / fix each site as appropriate?
```

Comment:
Who is @mtklein ? Does this code from upstream ?

Reply:
I bet that's me!

This comment and suppressed warning was in upstream Skia for a couple weeks, but removed in December 2019.  You can see the timeline at https://bugs.chromium.org/p/skia/issues/detail?id=9674.  I think we fixed whatever was bothering -Wclass-memaccess at the time.

Reply:
yes, this code from upstream, I just cherry pick minimal changes needed to compile in M1.

Reply:
Ah ok Github showed you were author. So you Co-authored the first commit and fully the second one ?

Does `xcrun esy` works ? and does our upstream mono/skia (We are fork of fork) compile fine ?

Reply:
@Et7f3 yes, that also works and it produces an arm64 library, previously it produces an x86 library.

The first attempt to compile our fork is actually merged mono SKIA to our repository, but that is too big to manage, and it also can't compile, not sure why. So, I try to build Google SKIA, and it compiled successfully. Based on that, I figure out how they do it. The changes on the first commit are just generated by applying a patch I created with their changes in gn/BUILD.gn file, the work should be credited to the Google team. The second commit is just putting back our changes to that file.